src/software/simulated_tests/terminating_validation_functions/ball_kicked_validation.cpp

```diff
@@ -0,0 +1,15 @@
+#include "software/simulated_tests/terminating_validation_functions/ball_kicked_validation.h"
+
+#include "software/logger/logger.h"
+
+
+void ballKicked(Angle angle, std::shared_ptr<World> world_ptr,
+                ValidationCoroutine::push_type& yield)
+{
+    while (!world_ptr->ball().hasBallBeenKicked(angle))
+    {
+        std::stringstream ss;
+        ss << angle;
+        yield("Ball was not kicked at the angle " + ss.str());
```

Comment:
Is `streamstream` required here? Can the angle not be printed by using `tostring` in the yield?

Reply:
We have the <<operator for the Angle class, but not std::to_string since we typically use the string to debug and log. I think this is the only way